base/applications/mspaint/winproc.cpp

```diff
@@ -575,6 +576,44 @@ LRESULT CMainWindow::OnSysColorChange(UINT nMsg, WPARAM wParam, LPARAM lParam, B
     return 0;
 }
 
+typedef ULONG (WINAPI *FN_MAPISendMailW)(LHANDLE, ULONG_PTR, lpMapiMessageW, FLAGS, ULONG);
+
+ULONG CMainWindow::OpenMailerWithAttachmentW(LPCWSTR pszFileName)
+{
+    if (!PathFileExistsW(pszFileName))
+    {
+        ::ExpandEnvironmentStringsW(L"%TEMP%\\Image.png", g_szTempFile, _countof(g_szTempFile));
+        if (!imageModel.SaveImage(g_szTempFile))
+        {
+            g_szTempFile[0] = UNICODE_NULL;
+            return MAPI_E_ATTACHMENT_NOT_FOUND;
+        }
+        pszFileName = g_szTempFile;
+    }
+
+    MapiFileDescW attachment = { 0 };
+    attachment.nPosition = (ULONG)-1;
+    attachment.lpszPathName = (LPWSTR)pszFileName;
```

Comment:
You need also to set the `lpszFileName` member of the `attachment` structure (see the screenshot attached).

Reply:
`MAPISendMailW` is Win8+. Are you OK?

Reply:
We are definitely not ok with breaking our MSpaint on 2k3sp2. I wouldn't block any optional feature though that would be only loaded at runtime when it detects a Win8 is present, and falling back gracefully otherwise.
Iirc even 2k3sp2/XP do offer a way to invoke mailto: via the shell somehow.

Reply:
Fallback to ANSI added.